File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;

        int more = 0;
        
        for (int i = 0; i < nums.size(); i++) {
            more = target - nums[i];

            if (mp.find(more) != mp.end()) {
                return { mp[more], i};
                
            }
            
            mp[nums[i]] = i;
            
        }
        return {-1, -1};
    }
};

File: 1-two-sum/two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;  // ✅ Use .length for arrays
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < n; i++) {
            int more = target - nums[i];
            if (map.containsKey(more)) {  // ✅ method is containsKey (not containKey)
                return new int[]{i, map.get(more)};  // ✅ Correct return syntax
            }
            map.put(nums[i], i);  // ✅ Key = nums[i], Value = i
        }
        return new int[]{-1, -1};  // ✅ Return default array
    }
}
